### PR TITLE
fix: check for numeric filtering results instead of boolean

### PR DESCRIPTION
**Pull request by ttvi**
Fix the missed out check in #651 for the filtering results of listing in bucketfile (should check for numeric values instead of boolean).

### CODE REVIEW

**Comment by ttvi** [comment]
Expectation: the commit for the unit test should fail and the commit for the fix should not.

File: lib/metadata/bucketfile/backend.js [hunk 1]
```diff
@@ -432,7 +432,7 @@ class BucketFileInterface {
             const stream = db.createReadStream(requestParams);
             stream
                 .on('data', e => {
-                    if (extension.filter(e) === false) {
+                    if (extension.filter(e) < 0) {
```
Comment (ThibaultRiviere) [inline lib/metadata/bucketfile/backend.js]: Shouldn't use the filter variable from arsenal ?

Comment (ttvi) [inline lib/metadata/bucketfile/backend.js]: Same answer from https://github.com/scality/MetaData/pull/1009/files#r107927401: no, metadata only cares if the value is `< 0`, it does not need to understand the specific values (-1, -2, ...) unless we map these values to specific actions in metadata.

File: tests/unit/metadata/bucketfile/backend.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    bucketfile.loadDBIfExists = (bucket, log, callback) => callback(null, db);
+    // eslint-disable-next-line no-unused-vars
+    const unRef = bucketfile.unRef;
+    bucketfile.unRef = () => {};
+    // remember to set the original functions back if having more tests
+
+    Object.keys(extensions).forEach(listingType => {
+        it(`listing max ${MAX_KEYS} keys using ${listingType}`, done => {
+            const params = { listingType, maxKeys: MAX_KEYS };
+            // assertion to check if extensions work with maxKeys is in Reader
```
Comment (jonathan-gramain) [inline tests/unit/metadata/bucketfile/backend.js]: I don't see the assertion you're referring to here.

Comment (jonathan-gramain) [inline tests/unit/metadata/bucketfile/backend.js]: Oh it must be the assertion in the `start()` function then.

**Comment by ttvi** [comment]
@ironman-machine hello

File: tests/unit/metadata/bucketfile/backend.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    // stub db to inspect the extensions
+    const db = {
+        createReadStream: () => new Reader(),
+    };
+
+    // stub functions and components
+    const logger = { info: () => {}, debug: () => {}, error: () => {} };
+    // eslint-disable-next-line no-unused-vars
+    const loadDBIfExists = bucketfile.loadDBIfExists;
```
Comment (LaurenSpiegel) [inline tests/unit/metadata/bucketfile/backend.js]: why is this defined?  To restore the function after the test?  If so, let's do that now in an after block.

File: tests/unit/metadata/bucketfile/backend.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    bucketfile.loadDBIfExists = (bucket, log, callback) => callback(null, db);
+    // eslint-disable-next-line no-unused-vars
+    const unRef = bucketfile.unRef;
+    bucketfile.unRef = () => {};
+    // remember to set the original functions back if having more tests
```
Comment (LaurenSpiegel) [inline tests/unit/metadata/bucketfile/backend.js]: Let's just do that clean up now.

File: tests/unit/metadata/bucketfile/backend.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        process.nextTick(() => this.start());
+    }
+
+    start() {
+        if (this.done) {
+            return null;
+        }
+        const i = this.index++;
+        // extensions should take into account maxKeys
+        // and should not filter more than that intented value
```
Comment (LaurenSpiegel) [inline tests/unit/metadata/bucketfile/backend.js]: typo -- intended

**Pull request merged**